### PR TITLE
Add audio decoding

### PR DESCRIPTION
**Pull request by charmoniumQ**
- Approve after: #84

### CODE REVIEW

**Review by apache8080 — DISMISSED** [review]

**Comment by mhuzai** [comment]
I am unable to test this. Checking out the branch results in:
`fatal: not a git repository: audio_decoding/libspatialaudio/../../.git/modules/audio_decoding/libspatialaudio`

**Review by JeffreyZh4ng — COMMENTED** [review]
LGTM. Illixr runs fine - verified that the component was being run

**Review by JeffreyZh4ng — DISMISSED** [review]

**Review by mhuzai — APPROVED** [review]
As discussed, I tested the gitmodules fix on my home machine and it worked fine. This PR is good to go.

**Comment by mhuzai** [comment]
@apache8080 brought up a good point that "audio_pipeline" is now misleading because it only contains encoding. One possible solution to this problem is to have just the `audio_pipeline` submodule, but have it internally launch two threads, one for encoding and one for playback (we should really name it playback and not decoding), just like ZED launches two threads. That would ensure that the term "audio_pipeline" still makes sense and we still get our two different threads. It would also remove the need for symlinks. Thoughts, @charmoniumQ?

**Comment by charmoniumQ** [comment]
I think that is cleaner and it only builds `libspatialaudio` once, so I implemented to that design.

File: configs/monado.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ plugins:
   - path: timewarp_gl/
   - path: debugview/
   - path: audio_pipeline/
+  - path: audio_decoding/
```
Comment (mhuzai) [inline configs/monado.yaml:10]:
```suggestion

```

**Review by mhuzai — COMMENTED** [review]
Wouldn't it be better to have an audio base class that both threads inherit from to minimize the duplication of code?

**Comment by charmoniumQ** [comment]
> Wouldn't it be better to have an audio base class that both threads inherit from to minimize the duplication of code?

That's a good point; I updated the implementation (actually two instances, one class).

**Review by mhuzai — APPROVED** [review]

**Review by mhuzai — APPROVED** [review]

**Review by apache8080 — APPROVED** [review]

**Pull request merged**